haskell_scotty/ClientRepo.hs

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE OverloadedStrings #-}
+module ClientRepo where
+
+import Client
+import DBConfig
+import Database.MySQL.Simple
+import Database.MySQL.Simple.Result
+import Database.MySQL.Simple.QueryResults
+
+all :: IO [Client]
+all = do
+  DBConfig user pass database <- readDBConfig
+  conn <- connect defaultConnectInfo {
+            connectUser = user,
+            connectPassword = pass,
+            connectDatabase = database }
+  clients <- query_ conn "SELECT * FROM clients"
+  return clients
+
+instance QueryResults Client where
+  convertResults [fa,fb,fc,fd,fe,ff,fg,fh,fi,fj,fk,fl] [va,vb,vc,vd,ve,vf,vg,vh,vi,vj,vk,vl] =
+    Client a b c d e f g h i j k l 
+      where a = convert fa va
+            b = convert fb vb
+            c = convert fc vc
+            d = convert fd vd
+            e = convert fe ve
+            f = convert ff vf
+            g = convert fg vg
+            h = convert fh vh
+            i = convert fi vi
+            j = convert fj vj
+            k = convert fk vk
+            l = convert fl vl
```

Comment:
Pretending that `v` was magically a tuple above, is there a way to apply a function to a tuple of arguments in Haskell? Something like https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply ?